catkit2/services/oceanoptics_spectrometer/oceanoptics_spectrometer.py

```diff
@@ -0,0 +1,189 @@
+        ImportError
+            If the spectrometer cannot be found.
+        '''
+
+        # Find the spectrometer
+        try:
+            self.spectrometer = Spectrometer.from_serial_number(self.serial_number)
+        except SeaBreezeError:
+            raise ImportError(f'OceanOptics: Could not find spectrometer with serial number {self.serial_number}')
+
+        # exctract attributes
+        self.model = self.spectrometer.model
+        self.pixels_number = self.spectrometer.pixels
+
+        # Define and set default exposure time.
+        self.exposure_time = self._exposure_time
+
+        # Create datastreams
+        self.spectra = self.make_data_stream('spectra', 'float32', [self.pixels_number], self.NUM_FRAMES)
+
+        self.is_saturating = self.make_data_stream('is_saturating', 'int8', [1], self.NUM_FRAMES)
+        self.is_saturating.submit_data(np.array([0], dtype='int8'))
+
+        # Create properties
+        def make_property_helper(name, read_only=False):
+            if read_only:
+                self.make_property(name, lambda: getattr(self, name))
+            else:
+                self.make_property(name, lambda: getattr(self, name), lambda val: setattr(self, name, val))
+
+        make_property_helper('exposure_time')
+        make_property_helper('wavelengths', read_only=True)
+
+    def main(self):
+        '''
+        The main function of the service.
+
+        This function is called when the service is started.
+        '''
+        while not self.should_shut_down:
+            intensities = self.take_one_spectrum()
+            self.spectra.submit_data(np.array(intensities, dtype='float32'))
+            self.sleep(self.interval)
+
+    def take_one_spectrum(self):
+        '''
+        Measure one spectrum and submit it to the data stream.
+        '''
+        intensities = self.spectrometer.intensities()
+
+        if np.max(intensities) ==  self.spectrometer.max_intensity:
+            self.is_saturating.submit_data(np.array([1], dtype='int8'))
+        else:
+            self.is_saturating.submit_data(np.array([0], dtype='int8'))
+
+        return intensities
+
+    @property
+    def wavelengths(self):
+        '''
+        The wavelengths of the spectrometer.
+
+        Returns:
+        --------
+        ndarray of float:
+            wavelengths of the spectrometer.
+        '''
+        return self.spectrometer.wavelengths()
+
+    @property
+    def exposure_time(self):
+        '''
+        The exposure time in microseconds.
+
+        Returns:
+        --------
+        int:
+            The exposure time in microseconds.
+        '''
+        return self._exposure_time
+
+    @exposure_time.setter
+    def exposure_time(self, exposure_time: int):
+        '''
+        Set the exposure time in microseconds.
+
+        if exposure_time < min allowed exp time: exposure_time =  min allowed exp time
+        if exposure_time > max allowed exp time: exposure_time =  max allowed exp time
```

Comment:
Can you write this out in words? The current docstring essentially just repeats the code and is a little hard to parse and/or unnecessary if the code says almost exactly the same thing. Detail though.

Reply:
done